extensions-core/namespace-lookup/src/main/java/io/druid/server/lookup/namespace/cache/OnHeapNamespaceExtractionCacheManager.java

```diff
@@ -100,11 +94,36 @@ public boolean delete(final String namespaceKey)
     final Lock lock = nsLocks.get(namespaceKey);
     lock.lock();
     try {
-      super.delete(namespaceKey);
-      return mapMap.remove(namespaceKey) != null;
+      return super.delete(namespaceKey) && mapMap.remove(namespaceKey) != null;
     }
     finally {
       lock.unlock();
     }
   }
+
+  @Override
+  protected void monitor(ServiceEmitter serviceEmitter)
+  {
+    long numEntries = 0;
+    long size = 0;
+    for (Map.Entry<String, ConcurrentMap<String, String>> entry : mapMap.entrySet()) {
+      final ConcurrentMap<String, String> map = entry.getValue();
+      if (map == null) {
+        LOG.debug("missing cache key for reporting [%s]", entry.getKey());
+        continue;
+      }
+      numEntries += map.size();
+      for (Map.Entry<String, String> sEntry : map.entrySet()) {
+        final String key = sEntry.getKey();
+        final String value = sEntry.getValue();
+        if (key == null || value == null) {
+          LOG.debug("Missing entries for cache key [%s]", entry.getKey());
```

Comment:
this is for debugging purposes, not for information purposes. I can move to Trace if you want. Basically I put it there so there is some debugging for why things are skipped.